### DISCUSSION TITLE
微信小程序端 yarn build:weapp 时出现错误

### DISCUSSION THREAD

**Discussion by hub607** [NervJS/taro#10799] [Q & A]
### taro版本为1.3.11

我是一个后端开发现在需要构建一个微信小程序项目
I am a back-end developer. Now I need to build a wechat applet project
初始化代码`yarn`时出现问题各种缺包问题 我尝试使用
There was a problem initializing the code 'yarn'. There are various package shortage problems. I try to use it

```
yarn add global @tarojs/mini-runner @tarojs/plugin-uglifyjs @tarojs/plugin-babel @tarojs/plugin-sass @tarojs/plugin-stylus babel-plugin-transform-class-properties babel-plugin-transform-decorators-legacy babel-plugin-transform-export-extensions babel-plugin-transform-object-rest-spread babel-preset-env --save-dev
```

成功初始化代码，在我使用`yarn build:weapp`时
Successful initialization code, When I use my `yarn build:weapp`
在node_modules\@tarojs\plugin-sass\dist\index.js 14:9 
TypeError: ctx.modifyWebpackChain is not a function
![image](https://user-images.githubusercontent.com/38717760/144403405-92cbaa5e-7621-4244-94fc-3006f5bcbe9d.png)

将这段代码注释以后出现
Comment this code to appear later
![image](https://user-images.githubusercontent.com/38717760/144403550-9620a0d3-764b-4b55-b2fb-4f2ed68577ed.png)

**Comment by luckyadam** [comment]
建议升级到最新的 3.x 版本

**Reply by hub607** [reply]
是个旧项目听说语法改动较大 尽量还是不升级吧

**Comment by 15380831711** [comment]
环境信息没贴出来，@tarojs/cli命令行工具版本和项目依赖的taro版本保持一致试试

**Reply by hub607** [reply]
cli命令行版本与项目依赖一致都是1.3.11    本地node 14.15.4   npm 6.14.10 yran 1.22.15

**Reply by 15380831711** [reply]
猜测是某些依赖自动升级导致，项目依赖lock文件没保留下来吗，尝试把所有依赖的版本锁定（写死）
